File: Source/TurnTracker.Data/Entities/Participant.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;

namespace TurnTracker.Data.Entities
{
    public class Participant : Entity
    {
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }

        public int UserId { get; set; }
        public User User { get; set; }

        public int ActivityId { get; set; }
        public Activity Activity { get; set; }

        public int TurnsNeeded { get; set; }
        public int TurnOrder { get; set; }

        public bool HasDisabledTurns { get; set; }

        public List<NotificationSetting> NotificationSettings { get; set; }
    }
}

File: Source/TurnTracker.Client/src/app/user/models/Participant.ts
import { NotificationSetting } from './NotificationSetting';

export enum VerificationStatus {
    None,
    Pending,
    Expired,
    Verified
}

export class Participant {
    public id: number;
    public userId: number;
    public name: string;
    public turnsNeeded: number;
    public hasDisabledTurns: boolean;
    public turnOrder: number;
    public notificationSettings: NotificationSetting[];
    public mobileNumberVerification: VerificationStatus;
    public emailVerification: VerificationStatus;
}
